src/main/python/sweep.py

```diff
@@ -327,21 +377,25 @@ def test_response_plot_data(sweep, plot_duration=0.1, num_baseline_samples=100):
     )
 
 
-def make_test_pulse_plot(sweep_number, time, voltage, previous=None, initial=None):
+def make_test_pulse_plot(sweep_number, time, voltage, previous=None, initial=None, step=1, labels=True):
     
     fig, ax = plt.subplots(figsize=DEFAULT_FIGSIZE)
 
     if initial is not None:
-        ax.plot(time, initial, linewidth=1, label=f"initial", color="green")
+        ax.plot(time[::step], initial[::step], linewidth=1, label=f"initial", color="green")
         
     if previous is not None:
-        ax.plot(time, previous, linewidth=1, label=f"previous", color="orange")
+        ax.plot(time[::step], previous[::step], linewidth=1, label=f"previous", color="orange")
     
-    ax.plot(time, voltage, linewidth=1, label=f"sweep {sweep_number}", color="blue")
+    ax.plot(time[::step], voltage[::step], linewidth=1, label=f"sweep {sweep_number}", color="blue")
 
-    ax.set_xlabel("time (s)", fontsize=PLOT_FONTSIZE)
-    ax.set_ylabel("membrane potential (V)", fontsize=PLOT_FONTSIZE)
-    ax.legend()
+    if labels:
+        ax.set_xlabel("time (s)", fontsize=PLOT_FONTSIZE)
+        ax.set_ylabel("membrane potential (V)", fontsize=PLOT_FONTSIZE)
```

Comment:
Units are (mV). I believe it would be sufficient to just use the unit label without naming the axis (i.e., "mV"), same for the x-axis ("s") - both for thumbnail and full figures. This is how they did it in the technical white paper on celltypes.brain-map.org